core/server/src/main/java/alluxio/web/UIWebServer.java

```diff
@@ -65,29 +69,36 @@ public UIWebServer(ServiceType service, InetSocketAddress address) {
     QueuedThreadPool threadPool = new QueuedThreadPool();
     int webThreadCount = Configuration.getInt(PropertyKey.WEB_THREADS);
 
-    mServer = new Server();
-    SelectChannelConnector connector = new SelectChannelConnector();
-    connector.setHost(address.getHostName());
-    connector.setPort(address.getPort());
-    connector.setAcceptors(webThreadCount);
-    mServer.setConnectors(new Connector[] {connector});
-
-    try {
-      mServer.getConnectors()[0].close();
-      mServer.getConnectors()[0].open();
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
-    }
-
     // Jetty needs at least (1 + selectors + acceptors) threads.
     threadPool.setMinThreads(webThreadCount * 2 + 1);
     threadPool.setMaxThreads(webThreadCount * 2 + 100);
-    mServer.setThreadPool(threadPool);
+
+    mServer = new Server(threadPool);
+
+    mServerConnector = new ServerConnector(mServer);
+    mServerConnector.setPort(mAddress.getPort());
+    mServerConnector.setHost(mAddress.getAddress().getHostAddress());
+
+    mServer.addConnector(mServerConnector);
```

Comment:
I don't see how we could use a utils method without having a race condition where a 3rd-party service takes the port we want before we bind to it. Both Jetty and Thrift bind the port during construction, not in the start method, so it doesn't seem too weird for us to do the same here. It makes it a lot easier to manage the port resetting situation when it's all done in one place.